File: api/src/main/java/fr/epf/speedycart/api/model/Product.java
package fr.epf.speedycart.api.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.Data;

import java.util.Date;

@Entity
@Data
public class Product {

        @Id 
        private Long productId;
        private String name;
        private Float unitPrice;
        private String description;
        private Long stock;
        private Date activeFrom;
        private Boolean deactivated;
        private Date deactivateFrom;
        private Float weight;
        private Float sizes;
        private Boolean forAdults;
        private Long shopId;
}

File: app/app/src/main/java/fr/epf/min1/speedycart/data/Product.kt
package fr.epf.min1.speedycart.data

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.time.LocalDateTime

@Parcelize
data class Product(
    val id: Long?,
    val name: String,
    val unitPrice: Double,
    val description: String?,
    val stock: Int,
    val activeSince: LocalDateTime?,
    val disableSince: LocalDateTime?,
    val weight: Double,
    val sizes: Double,
    val forAdults: Boolean,
    val shop: Shop
) : Parcelable {
    companion object {
        fun generateListProduct(nb: Int = 10): List<Product> =
            (1..nb).map {
                Product(
                    it.toLong(),
                    "Pates Panza $it",
                    1.24,
                    null,
                    50 + it,
                    null,
                    null,
                    500.0,
                    1.0,
                    false,
                    Shop.generate1Shop()
                )
            }
    }
}